### ISSUE TITLE
Update docker-py version to support docker logging drivers in Docker 1.8

### ISSUE THREAD

**Issue by willowandcreek** [rancher/rancher#2313]
Hi,

Looks like the new logging drivers in 1.8 aren't supported. For example when I try to enable GELF as a logging driver I get the following in my agent log:

```
2015-10-12T03:42:15.446170281Z 2015-10-12 03:42:15,445 ERROR agent [139987274000816] [event.py:111] b2187b11-130c-4983-a8b7-ea3e54763901 : Unknown error
2015-10-12T03:42:15.446204421Z Traceback (most recent call last):
2015-10-12T03:42:15.446215345Z   File "/var/lib/cattle/pyagent/cattle/agent/event.py", line 94, in _worker_main
2015-10-12T03:42:15.446223213Z     resp = agent.execute(req)
2015-10-12T03:42:15.446229015Z   File "/var/lib/cattle/pyagent/cattle/agent/__init__.py", line 15, in execute
2015-10-12T03:42:15.446235466Z     return self._router.route(req)
2015-10-12T03:42:15.446250867Z   File "/var/lib/cattle/pyagent/cattle/plugins/core/event_router.py", line 13, in route
2015-10-12T03:42:15.446257189Z     resp = handler.execute(req)
2015-10-12T03:42:15.446262519Z   File "/var/lib/cattle/pyagent/cattle/agent/handler.py", line 34, in execute
2015-10-12T03:42:15.446268155Z     return method(req=req, **req.data.__dict__)
2015-10-12T03:42:15.446273440Z   File "/var/lib/cattle/pyagent/cattle/plugins/docker/compute.py", line 508, in instance_activate
2015-10-12T03:42:15.446279098Z     self._do_instance_activate(instance, host, progress)
2015-10-12T03:42:15.446284368Z   File "/var/lib/cattle/pyagent/cattle/plugins/docker/compute.py", line 566, in _do_instance_activate
2015-10-12T03:42:15.446290121Z     create_config['host_config'] = create_host_config(**start_config)
2015-10-12T03:42:15.446295570Z   File "/var/lib/cattle/pyagent/dist/docker/utils/utils.py", line 480, in create_host_config
2015-10-12T03:42:15.446301137Z     log_config = LogConfig(**log_config)
2015-10-12T03:42:15.446306483Z   File "/var/lib/cattle/pyagent/dist/docker/utils/types.py", line 27, in __init__
2015-10-12T03:42:15.446312060Z     ', '.join(self.types._values)
2015-10-12T03:42:15.446317340Z ValueError: LogConfig.type must be one of (json-file, syslog, none)
```

My environment is:

```
Rancher 	v0.40.0
Cattle 	v0.97.0
User Interface 	v0.54.0
Rancher Compose 	beta/latest
Docker: 1.8.2
```

The way I am trying to enable the log driver is via docker-compose.yml:

```yml
  external_links:
  - infra/graylog:graylog
  log_driver: "gelf"
  log_opt:
    address: "udp://graylog:12201"
```

**Comment by deniseschannon** [comment]
Could you share your full docker-compose.yml? I'm not familiar with logging drivers, but would like to see it.

**Comment by willowandcreek** [comment]
Hi,

Here is the docker-compose:

```yml
web:
  restart: always
  tty: true
  image: reg.local/test/hello-world:1.0.0
  stdin_open: true
  links:
  - graylog
  log_driver: "gelf"
  log_opt:
    address: "udp://graylog:12201"

graylog:
  restart: always
  tty: true
  image: graylog2/allinone
  environment:
    GRAYLOG_TIMEZONE: GMT
  stdin_open: true
  expose:
  - 9000
  - 12201
  - 12201/udp

```

This works on docker 1.8 via the command line. For example: ```docker run --log-driver=gelf --log-opt gelf-address=udp://graylog_ip:12201``` as shown at https://docs.docker.com/reference/logging/overview/#specify-gelf-options. Support was also added to the docker-compose as shown at https://docs.docker.com/compose/yml/#log-driver

**Comment by ibuildthecloud** [comment]
@willowandcreek We do support gelf and other logging drivers.  I tried your compose file and ran into a couple issues.

1) The log-opt is `gelf-address` and not `address`.  So basically the following

```yaml
log_opt:
    gelf-address: "udp://graylog:12201"
```

2) You can't reference the name "graylog".  Even though you have linked the container graylog, the Docker daemon is the one that resolves the DNS name "graylog" and not the container.  So the name "graylog" is not going to be available on the host.  You need to point to some other address.

**Comment by willowandcreek** [comment]
@ibuildthecloud Thanks for the feedback. Could I check if this support is there in v0.41.0? I just tried the following

- Move graylog onto a standalone server
- Upgraded my rancher from v0.40.0 to v0.41.0
- Ran my image via docker cli and was able to see the log messages on the graylog server
- Ran a plain ubuntu image using Infrastructure/Containers and was able to communicate with the graylog server (tried this just to rule out network comm issues)
- Ran my image via rancher-compose and had the same error.

My updated docker-compose:

```yml
web:
  restart: always
  tty: true
  image: reg.local/test/hello-world:1.0.0
  stdin_open: true
  log_driver: "gelf"
  log_opt:
    gelf-address: "udp://10.65.1.106:12201"
```

where, 10.65.1.106 is a regular ubuntu server where my graylog server is running. Launching my stack resulted in the same stack trace:

```
2015-10-16T03:46:13.856180842Z Traceback (most recent call last):
2015-10-16T03:46:13.856191010Z   File "/var/lib/cattle/pyagent/cattle/agent/event.py", line 94, in _worker_main
2015-10-16T03:46:13.856198917Z     resp = agent.execute(req)
2015-10-16T03:46:13.884067034Z   File "/var/lib/cattle/pyagent/cattle/agent/__init__.py", line 15, in execute
2015-10-16T03:46:13.884101704Z     return self._router.route(req)
2015-10-16T03:46:13.884124600Z   File "/var/lib/cattle/pyagent/cattle/plugins/core/event_router.py", line 13, in route
2015-10-16T03:46:13.884133550Z     resp = handler.execute(req)
2015-10-16T03:46:13.884139580Z   File "/var/lib/cattle/pyagent/cattle/agent/handler.py", line 34, in execute
2015-10-16T03:46:13.884145887Z     return method(req=req, **req.data.__dict__)
2015-10-16T03:46:13.884151922Z   File "/var/lib/cattle/pyagent/cattle/plugins/docker/compute.py", line 508, in instance_activate
2015-10-16T03:46:13.884158079Z     self._do_instance_activate(instance, host, progress)
2015-10-16T03:46:13.884163927Z   File "/var/lib/cattle/pyagent/cattle/plugins/docker/compute.py", line 566, in _do_instance_activate
2015-10-16T03:46:13.884170067Z     create_config['host_config'] = create_host_config(**start_config)
2015-10-16T03:46:13.884175977Z   File "/var/lib/cattle/pyagent/dist/docker/utils/utils.py", line 480, in create_host_config
2015-10-16T03:46:13.884182091Z     log_config = LogConfig(**log_config)
2015-10-16T03:46:13.884187791Z   File "/var/lib/cattle/pyagent/dist/docker/utils/types.py", line 27, in __init__
2015-10-16T03:46:13.884193744Z     ', '.join(self.types._values)
2015-10-16T03:46:13.884199358Z ValueError: LogConfig.type must be one of (json-file, syslog, none)
```

**Comment by deniseschannon** [comment]
Can you do a `rancher-compose -v` to let me know the exact version of Rancher compose that you are running? With v0.41.0, it should be used with v0.4.1.

**Comment by willowandcreek** [comment]
Yep its v0.4.1. I always download the rancher-compose when I update my rancher server version.

```
~ $ rancher-compose -v
rancher-compose version v0.4.1
```

**Comment by willowandcreek** [comment]
Looks like this was an issue with docker-py and was fixed in version 1.5.0 (http://stackoverflow.com/questions/32479530/using-docker-compose-with-gelf-log-driver).
